keepassxc-browser/content/icon-handler.js

```diff
@@ -0,0 +1,238 @@
+'use strict';
+
+/**
+ * @Object kpxcIcons
+ * Icon handling.
+ */
+const kpxcIcons = {};
+kpxcIcons.icons = [];
+kpxcIcons.iconTypes = { USERNAME: 0, PASSWORD: 1, TOTP: 2 };
+
+// Adds an icon to input field
+kpxcIcons.addIcon = async function(field, iconType) {
+    if (!field || iconType < 0 || iconType > 2) {
```

Comment:
since you define iconTypes, why not just check if the value is in the set? A debug warning should be thrown here as well since there is clearly an issue in the code if iconType is not valid.